### DISCUSSION TITLE
How does the docker version skip the launcher?

### DISCUSSION THREAD

**Discussion by zacha81** [netbrain/zwift#66] [Q&A] [answered]
I was wondering if there was a way to bypass it on Windows too but I could not figure out what you are doing differently. I can run Zwift from ZwiftLauncher.exe or ZwiftApp.exe .. no matter what it will bring up the Browser based launcher window asking for my credentials. Can anyone explain what is diffent on docker/wine?

**Comment by quietvoid** [comment, marked as answer]
See the entrypoint script: https://github.com/netbrain/zwift/blob/master/entrypoint.sh#L76
The launcher is started to tray with `SilentLaunch`, then the process PID is used to start the game with `runfromprocess`.

For authentication the token is fetched by this script: https://github.com/netbrain/zwift/blob/master/zwift-auth.sh
And passed to the `ZwiftApp.exe` start with `--token ...`

**Reply by zacha81** [reply]
thank you
